### PR TITLE
restructure `CalibTracker/SiPixelESProducers` fake conditions `ESProducer`s

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

Full restructuring of `CalibTracker/SiPixelESProducers` fake conditions `ESProducer`s
- fold back header files into plugin implementation
- modernize code, add `fillDescriptions`
- add stub of `test_catch2_ESSources`
- make plugins in CalibTracker/SiPixelESProducers/test self-contained

improve `SiPixelTemplateDBObjectReader`
- add `SiPixelTemplateDBObjectReader_cfg.py` test in `createDBObject` test

#### PR validation:

   * `scram b runtests_createDBObjects` runs fine
   * `scram b runtests_testSiPixelESSources` runs fine

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by perrotta** [comment]
+1

File: CalibTracker/SiPixelESProducers/plugins/SiPixelFakeGainESSource.cc [hunk 1]
```diff
@@ -12,17 +12,51 @@
 */
 //
 // Original Author:  Vincenzo Chiochia
-//         Created:  Fri Apr 27 12:31:25 CEST 2007
+//         Created:  Tue 8 12:31:25 CEST 2007
```
[the remark this answers is not in the document]
Comment (mmusich) [reply CalibTracker/SiPixelESProducers/plugins/SiPixelFakeGainESSource.cc:15]: that's the date that was in the header file (different from the .cc).

[the remark this answers is not in the document]
Comment (ferencek) [reply CalibTracker/SiPixelESProducers/plugins/SiPixelFakeGainESSource.cc:15]: it is then probably better to leave the date from the .cc file than to replace it with an undefined one (missing month)

[the remark this answers is not in the document]
Comment (mmusich) [reply CalibTracker/SiPixelESProducers/plugins/SiPixelFakeGainESSource.cc:15]: Feel free to that in a follow-up PR.

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**